src/server.h

```diff
@@ -2473,6 +2473,7 @@ int moduleGetCommandKeysViaAPI(struct redisCommand *cmd, robj **argv, int argc,
 int moduleGetCommandChannelsViaAPI(struct redisCommand *cmd, robj **argv, int argc, getKeysResult *result);
 moduleType *moduleTypeLookupModuleByID(uint64_t id);
 moduleType *moduleTypeLookupModuleByName(const char *name);
+moduleType *moduleTypeLookupModuleByNameIgnoreCase(const char *name);
```

Comment:
i might be missing something, but i don't see this new function being used anywhere.
don't you need to use it in getObjectTypeByName? or maybe better, let scanGenericCommand pass an `ignore_case` argument to it, and make it more explicit.

let's add a simple test for this (would have uncovered this oversight which we almost missed twice, and protect future changes from breaking it).

Reply:
Sorry, I was distracted and caused this careless question. I found it difficult to avoid string comparisons in redis7.2, using numeric comparisons still cause a breaking change. Such as tow module types,"A12345678" and "a12345678", in the past server would return all keys belonging both "A12345678" and "a12345678", but now if we use interger compareison, the server only return the keys belonging either "A12345678" or "a12345678".

Reply:
you mean if there are two modules with the same name (and only character case difference)?
i think we can overlook this (i don't think there's such a case in reality).
what's important IMO is that if a module is named "Abcd12345" we can find it with "abcd12345" and "ABCD12345" too.
so i think we can keep the integer compare, as long as we look it up with ignore_case.

Reply:
i'm also confused by your last comment. if you wanted to cover that case that i'm willing to give up, you should have kept the string compare, but looking at the code i see you still do integer compare...
so either way, let's add a test to cover this (without one module and several searches, not with two modules with similar name).

side note: i'm also willing to bring back the string compare for 7.2 and replace it in 8.0, but it seems to me that with small effort we can keep the integer compare too.

Reply:
I add a test case using a case insensitive scan for the module type, not sure if it is what you wanted